server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJmapTestExtension.java

```diff
@@ -39,47 +38,141 @@
 import org.junit.jupiter.api.extension.ParameterContext;
 import org.junit.jupiter.api.extension.ParameterResolutionException;
 import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.function.ThrowingConsumer;
 import org.junit.rules.TemporaryFolder;
 
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
 import com.google.inject.Module;
 
-class CassandraJmapTestExtension implements BeforeAllCallback, BeforeEachCallback, AfterAllCallback, AfterEachCallback, ParameterResolver {
+public class CassandraJmapTestExtension implements BeforeAllCallback, BeforeEachCallback, AfterAllCallback, AfterEachCallback, ParameterResolver {
+
+    public static class Builder {
+
+        public static Builder withDefaultFromModules(Module... modules) {
+            return builder()
+                .additionalModules(modules)
+                .extensions(EMBEDDED_ES_EXTENSION_ONLY);
+        }
+
+        private Module coreModule;
+        private ImmutableList<Module> additionalModules;
+        private ImmutableList<GuiceModuleTestExtension> extensions;
+        private boolean ignoreEach;
+
+        public Builder() {
+            this.coreModule = ALL_BUT_JMX_CASSANDRA_MODULE;
+            this.additionalModules = DEFAULT_MODULES;
+            this.extensions = EMPTY_EXTENSIONS;
+            this.ignoreEach = false;
+        }
+
+        public Builder coreModule(Module coreModule) {
+            this.coreModule = coreModule;
+            return this;
+        }
+
+        public Builder modules(Module... additionalModules) {
```

Comment:
additionalModules ?

Reply:
oh, this method replace the additionalModules but additionalModules don't. That's too complicated.